payload/builder/types.go

```diff
@@ -31,11 +31,11 @@ import (
 	statedb "github.com/berachain/beacon-kit/state-transition/core/state"
 )
 
-type PayloadCache[RootT, SlotT any] interface {
-	Get(slot SlotT, stateRoot RootT) (engineprimitives.PayloadID, bool)
-	Has(slot SlotT, stateRoot RootT) bool
-	Set(slot SlotT, stateRoot RootT, pid engineprimitives.PayloadID)
-	UnsafePrunePrior(slot SlotT)
+type PayloadCache interface {
+	GetAndEvict(slot math.Slot, stateRoot common.Root) (engineprimitives.PayloadID, bool)
```

Comment:
Usually I don't naming naming things `SomethingANDSomething` but here it seems appropriate?
- Evict only would be misleading as this is really a get that consumes the retrieved payload
- Get would not convey the fact that retrieved item is consumed
- `Pop` comes to mind? Maybe `Push` and `Pop` instead of `Set/Get`?

I think we will end up having this cache removed and having a single payloadID "pending". So GetAndEvict is totally fine for me now